pylearn2/linear/cudnn2d.py

```diff
@@ -0,0 +1,203 @@
+import functools
+import numpy as np
+
+from theano.sandbox.cuda.dnn import GpuDnnConv, GpuDnnConvDesc
+from theano.sandbox.cuda.basic_ops import gpu_contiguous
+
+from pylearn2.packaged_dependencies.theano_linear.conv2d \
+    import Conv2d as OrigConv2D
+
+from pylearn2.linear.linear_transform import LinearTransform as P2LT
+from pylearn2.utils import sharedX
+from pylearn2.utils.rng import make_np_rng
+
+
+default_seed = [2012, 11, 6, 9]
+default_sparse_seed = [2012, 11, 6]
+
+
+class Cudnn2D(OrigConv2D):
+    """
+    Wrapper on the Theano Cudnn op.
+
+    Parameters
+    ----------
+    filters : Theano shared variable
+        4D-tensor of shape (out channels, in channels, rows, cols)
+    batch_size : int
```

Comment:
Uhm....cudnn is this file, I am confused. :) 

batch_size is only used to compute the img_shape parameter passed to the _original Conv2D_ (`pylearn2.packaged_dependencies.theano_linear.conv2d`) constructor. The exact same thing is done in `pylearn2.linear.conv2d`.